### ISSUE TITLE
Wizard: improve a11y and keyboard navigation

### ISSUE THREAD

**Issue by mayank99** [iTwin/iTwinUI-react#277]
Wizard should have keyboard navigation similar to HorizontalTabs, and some aria attributes.

**Comment by mayank99** [comment]
@veekeys I thought this would be a straightforward task but apparently not 😄 

1. Not sure what aria attributes to add (we already have `aria-current`). There is no conventional "wizard" or "stepper" guidelines in W3. This is the only thing I found: https://www.w3.org/WAI/tutorials/forms/multi-page/#using-step-by-step-indicator 
2. Not sure how keyboard navigation would even work. We don't allow focusing "future" steps. And we probably shouldn't use `auto` focus-activation because it might be destructive (as the wizard content usually has user input). Also [W3 recommends focusing the first element](https://www.w3.org/TR/wai-aria-practices-1.1/#:~:text=move%20focus%20to%20the%20starting%20point%20for%20that%20action) inside wizard content whenever a new step is activated.

**Comment by bentleyvk** [comment]
> @veekeys I thought this would be a straightforward task but apparently not 😄
> 
> 1. Not sure what aria attributes to add (we already have `aria-current`). There is no conventional "wizard" or "stepper" guidelines in W3. This is the only thing I found: https://www.w3.org/WAI/tutorials/forms/multi-page/#using-step-by-step-indicator

I see that `aria-current` can also have value `step` which might be a better fir here:
![image](https://user-images.githubusercontent.com/36186912/130575876-9f57e943-512d-4944-b03c-62a1a798911d.png)


> 2. Not sure how keyboard navigation would even work. We don't allow focusing "future" steps. And we probably shouldn't use `auto` focus-activation (like in Tabs) because it might be destructive (as the wizard content usually has user input). Also [W3 recommends focusing the first element](https://www.w3.org/TR/wai-aria-practices-1.1/#:~:text=move%20focus%20to%20the%20starting%20point%20for%20that%20action) inside wizard content whenever a new step is activated.

I don't think we should allow to focus future steps as it usually is done thought some actions inside `Wizard`. 
For focusing first element we could use Vykintas helper function from his focus trap branch to find first focusable element. But I am not sure about this as devs might want to focus different elements.
